tool/tsh/db.go

```diff
@@ -70,20 +70,25 @@ func onListDatabases(cf *CLIConf) error {
 	}
 	defer proxy.Close()
 
-	databases, err := proxy.FindDatabasesByFiltersForCluster(cf.Context, *tc.DefaultResourceFilter(), profile.Cluster)
+	cluster := profile.Cluster
+	if cf.SiteName != "" {
+		cluster = cf.SiteName
```

Comment:
Any chance we could get a test?

Is it worth following profile.Cluster around to check for similar bugs?

Reply:
was hesitant about UT because it requires two clusters. End up adding it and I do feel good about it. thanks for the suggestion.

The issue only applies to commands that support `--cluster` flag and is only broken because of my recent change. I've tested `--cluster` for other `tsh db` commands and they are still good.

Reply:
Looking good, thanks!